### PR TITLE
Password change and reset throw 400

### LINKED ISSUE #2391: Password change and reset endpoints throw 500 instead of returning error messages
## Description
* Try updating a password with a common password such as `password`.
* The /password_reset and /password_change endpoints throw a 500
* They should return a 400 with formatted error messages

Related: 
* https://github.com/centerofci/mathesar/issues/1673

### PR DESCRIPTION
**Pull request by Hatilar420**
Fixes #2391

# Overview
1. The `/password_reset` and `/password_change` endpoints throw a 400 formatted response when the password is changed or reset with a common password

**Screenshots**

1. `/password_reset` formatted response.
image

2. `/password_change` formatted response.

image

## Checklist
 
- [X] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [X] My pull request targets the `master` branch of the repository
- [X] My commit messages follow [best practices][best_practices].
- [X] My code follows the established code style of the repository.
- [X] I added tests for the changes I made (if applicable).
- [X] I added or updated documentation (if applicable).
- [X] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by kgodey** [comment]
@Hatilar420 thanks for your contribution! this might take us a couple of weeks to get to since we're focused on launching our first release right now.

**Comment by Hatilar420** [comment]
sure np @kgodey.

File: mathesar/api/exceptions/mixins.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def _run_validator(self, validator, field, message):
                 args.append(field)
             validator(self.initial_data[field.field_name], *args)
         except (DjangoValidationError, RestValidationError) as err:
-            err_message = err.detail[0] if hasattr(err, 'detail') else err.message
+            err_message = err.detail[0] if hasattr(err, 'detail') else err.messages[0]
```
Comment (silentninja) [inline mathesar/api/exceptions/mixins.py:84]: - Django error objects sometimes contain only the `message` field. So `err.message` is sometimes valid too. So please revert its removal. In order to support various types of errors,  you would need to write the logic for each case.
- Also make sure to check if the `messages` array is not empty before fetching the first error object.

**Review by silentninja — CHANGES_REQUESTED** [review]
Thanks for the PR @Hatilar420. The error is captured correctly but I have a few concerns which are commented on in detail below, these need to be addressed before the PR is merged.

File: mathesar/api/exceptions/mixins.py (hunk 1 above)
Comment (Hatilar420) [reply mathesar/api/exceptions/mixins.py:84]: Valid points @silentninja, Will implement the suggested changes.

**Comment by Hatilar420** [comment]
Hi @silentninja, I have implemented the changes you requested for. I have some doubts.
- What should value should we assign to the `err_message` variable if the `messages` array were to be empty (Which is really unlikely unless and until there is some change in `DjangoValidationError`, `RestValidationError` by the Django team)
- What should be the default value for `err_message`?

**Review by silentninja — APPROVED** [review]

**Comment by silentninja** [comment]
> What should value should we assign to the err_message variable if the messages array were to be empty (Which is really unlikely unless and until there is some change in DjangoValidationError, RestValidationError by the Django team)
What should be the default value for err_message?

The above scenario should not likely occur, so we should leave it as is and get a 500 error thrown than hiding the problem

**Pull request merged**